shared/hooks/useDuckdam.ts

```diff
@@ -9,19 +10,30 @@ const useDuckdam = () => {
         const newDuckDam = {
             ...randomNewDuckDam(politeLevel),
         };
-
-        const response = await fetch('/api/duckdam/add', {
-            method: 'POST',
-            body: JSON.stringify(newDuckDam),
-            headers: {
-                'Content-Type': 'application/json',
-            },
-        });
-
-        const id = await response.json();
-        if (response.ok) {
-            setTimeout(() => router.push(`/result/${id}`), 3000);
-        }
+        // const response = await fetch('/api/duckdam/add', {
+        //     method: 'POST',
+        //     body: JSON.stringify(newDuckDam),
+        //     headers: {
+        //         'Content-Type': 'application/json',
+        //     },
+        // });
+
+        // const id = await response.json();
+        // if (response.ok) {
+        //     setTimeout(() => router.push(`/result/${id}`), 3000);
+        // }
+
+        const url = '/api/duckdam/add';
+        const dataDuckdam = newDuckDam;
+        axios
+            .post(url, dataDuckdam, {
+                headers: {
+                    'Content-Type': 'application/json',
+                },
+            })
+            .then(({ data }) =>
+                setTimeout(() => router.push(`/result/${data}`), 3000)
+            );
```

Comment:
요기도 Promise 문법을 개선해서 `async`, `await` 과 구조분해할당은 같이 사용하면 좀 더 간결해질 것 같네용!
```suggestion
        const url = '/api/duckdam/add';
        const { data, status } = await axios.post(url, newDuckDam, {
            headers: {
                'Content-Type': 'application/json',
            },
        });

        if (status === 200) {
            setTimeout(() => router.push(`/result/${data}`), 3000);
        }
```